### DISCUSSION TITLE
Sum all json

### DISCUSSION THREAD

**Discussion by PortaCapenaPL** [datalust/seq-tickets#1987] [Help]
Hello Team!
I have the following log entry:
![image](https://github.com/datalust/seq-tickets/assets/76167302/eafdd5c6-b592-4b7e-b64b-09885a115c48)
When I do 
```
select sum(my_json[0]['GridUsageW'])
from stream
where run_time = '2023-10-13T15:35:42.117840' group by my_json[0]['DateTime']
```
I get a nice table: 
<img width="684" alt="image" src="https://github.com/datalust/seq-tickets/assets/76167302/88e4868a-6410-4640-a520-664795aec17f">

But can I group all my array fields by date? 
I tried:
```
select sum(my_json[?]['GridUsageW'])
from stream
where run_time = '2023-10-13T15:35:42.117840' group by my_json[?]['DateTime']
```
But it returns only nulls.

What I want to achieve in the end is to sum all json values of `GridUsageW` key in this log entry. Is it possible?

**Comment by nblumhardt** [comment]
Hi! Yes, this is a good use case for `lateral unnest`:

```sql
select sum(item.GridUsageW)
from stream
lateral unnest(my_json) as item
where run_time = '2023-10-13T15:35:42.117840'
group by item.DateTime
```

Let me know if this does the trick!
